doc/classes/NavigationObstacle2D.xml

```diff
@@ -1,42 +1,71 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<class name="NavigationObstacle2D" inherits="Node" is_experimental="true" version="4.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<class name="NavigationObstacle2D" inherits="Node2D" is_experimental="true" version="4.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
-		2D Obstacle used in navigation for collision avoidance.
+		2D Obstacle used in navigation to constrain avoidance controlled agents outside or inside an area.
 	</brief_description>
 	<description>
-		2D Obstacle used in navigation for collision avoidance. The obstacle needs navigation data to work correctly. [NavigationObstacle2D] is physics safe.
-		Obstacles [b]don't[/b] change the resulting path from the pathfinding, they only affect the navigation agent movement in a radius. Therefore, using obstacles for the static walls in your level won't work because those walls don't exist in the pathfinding. The navigation agent will be pushed in a semi-random direction away while moving inside that radius. Obstacles are intended as a last resort option for constantly moving objects that cannot be (re)baked to a navigation mesh efficiently.
+		2D Obstacle used in navigation to constrain avoidance controlled agents outside or inside an area. The obstacle needs a navigation map and outline vertices defined to work correctly. [NavigationObstacle2D] is physics safe.
```

Comment:
Same here. Also, either change "physics safe" to "physics-safe", or explain it in full, like "Obstacle avoidance is computed before physics, so pathfinding information can be used in the physics step safely."

Reply:
I think this was just a sloppy copy from the NavigationAgent and should be removed. The old NavigationObstacle3D had the linear_velocity auto set for the obstacle fake agent while the NavigationObstacle2D had nothing similar so it was never really a thing for most uses.